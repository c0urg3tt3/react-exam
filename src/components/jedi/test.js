import React from 'react'
import { shallow } from 'enzyme'

import Jedi from './component'

import {
  expectJediListLoader,
  expectJediError,
  expectJediListEmpty,
  expectJediQuote
} from '../../utils/expectedComponents'

describe('[component] Jedi', () => {
  it('should render component', () => {
    const wrapper = shallow(<Jedi/>)

    expect(wrapper.exists()).toEqual(true)
    expect(wrapper.is('div')).toEqual(true)
    expect(wrapper.hasClass('jedi')).toEqual(true)

    expectJediError({wrapper, exist: false})
    expectJediListEmpty({wrapper})
    expectJediListLoader({wrapper, exist: false})
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should call fetchJedi when componentWillMount', () => {
    const mockFetchJedi = jest.fn()
    const wrapper = shallow(<Jedi fetchJedi={mockFetchJedi}/>)

    expect(mockFetchJedi.mock.calls.length).toBe(1);
  })

  it('should render fetching error', () => {
    const errorMessage = "death star destroyed !"
    const wrapper = shallow(<Jedi/>)

    expect(wrapper.hasClass('error')).toEqual(false)
    expectJediError({ wrapper: wrapper.find('JediError'), exist: false})
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({
      isFetchingError: true,
      fetchingError: errorMessage
    })

    expect(wrapper.hasClass('error')).toEqual(true)
    expectJediError({
      wrapper: wrapper.find('JediError'),
      isFetchingError: true,
      errorMessage
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render boundary error', () => {
    const errorMessage = "death star destroyed !"
    const wrapper = shallow(<Jedi/>)

    expect(wrapper.hasClass('error')).toEqual(false)
    expectJediError({ wrapper: wrapper.find('JediError'), exist: false})
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setState({
      isBoundaryError: true,
      boundaryError: errorMessage
    })

    expect(wrapper.hasClass('error')).toEqual(true)
    expectJediError({
      wrapper: wrapper.find('JediError'),
      isBoundaryError: true,
      errorMessage
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render children as a function', () => {
    const jedies = [
      {id: 42, name: 'Jar Jar Bings'},
      {id: 1337, name: 'Anakin Skywalkers'}
    ]
    const wrapper = shallow(
      <Jedi jedies={jedies}>
      { ({jedies}) => {
        return jedies.map(((jedi, index) => (
          <div key={index} className="jedi-list-item">
            {jedi.name}
          </div>
        )))
      }}
      </Jedi>
    )

    expect(wrapper.find('.jedi-list-item')).toHaveLength(2)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
