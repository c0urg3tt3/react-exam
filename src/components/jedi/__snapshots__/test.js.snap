// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] Jedi should render boundary error 1`] = `"<div class=\\"jedi\\"><form id=\\"jedi-form\\" class=\\"jedi-form\\"><div class=\\"jedi-form-field\\"><label class=\\"jedi-form-field-label\\" for=\\"jedi-form-field_name\\">Add a Jedi to the Council</label><input class=\\"jedi-form-field-input\\" id=\\"jedi-form-field_name\\" name=\\"jedi-form-field_name\\" value=\\"\\" placeholder=\\"name\\"/></div><button class=\\"jedi-form-button-submit\\" type=\\"submit\\" title=\\"[disabled] Submit your padawan to the Council\\" disabled=\\"\\"><img alt=\\"app logo\\" src=\\"logo_jedi_order.svg\\" class=\\"app-logo\\"/></button></form><div class=\\"jedi-list-empty\\"><div class=\\"jedi-quote\\"><p>Looks like someone executed Order 66 !</p><blockquote><p>Master sky walker there is too many of them what are we going to do?</p><footer>Padawan</footer></blockquote></div></div></div>"`;

exports[`[component] Jedi should render boundary error 2`] = `"<div class=\\"jedi error\\"><div class=\\"jedi-error\\"><div class=\\"jedi-quote\\"><p>death star destroyed !</p><blockquote><p>I felt a great disturbance in the Force, as if millions of voices suddenly cried out in terror and were suddenly silenced. I fear something terrible has happened.</p><footer>Obi-Wan Kenobi</footer></blockquote></div></div><form id=\\"jedi-form\\" class=\\"jedi-form\\"><div class=\\"jedi-form-field\\"><label class=\\"jedi-form-field-label\\" for=\\"jedi-form-field_name\\">Add a Jedi to the Council</label><input class=\\"jedi-form-field-input\\" id=\\"jedi-form-field_name\\" name=\\"jedi-form-field_name\\" value=\\"\\" placeholder=\\"name\\"/></div><button class=\\"jedi-form-button-submit\\" type=\\"submit\\" title=\\"[disabled] Submit your padawan to the Council\\" disabled=\\"\\"><img alt=\\"app logo\\" src=\\"logo_jedi_order.svg\\" class=\\"app-logo\\"/></button></form><div class=\\"jedi-list-empty\\"><div class=\\"jedi-quote\\"><p>Looks like someone executed Order 66 !</p><blockquote><p>Master sky walker there is too many of them what are we going to do?</p><footer>Padawan</footer></blockquote></div></div></div>"`;

exports[`[component] Jedi should render children as a function 1`] = `"<div class=\\"jedi\\"><form id=\\"jedi-form\\" class=\\"jedi-form\\"><div class=\\"jedi-form-field\\"><label class=\\"jedi-form-field-label\\" for=\\"jedi-form-field_name\\">Add a Jedi to the Council</label><input class=\\"jedi-form-field-input\\" id=\\"jedi-form-field_name\\" name=\\"jedi-form-field_name\\" value=\\"\\" placeholder=\\"name\\"/></div><button class=\\"jedi-form-button-submit\\" type=\\"submit\\" title=\\"[disabled] Submit your padawan to the Council\\" disabled=\\"\\"><img alt=\\"app logo\\" src=\\"logo_jedi_order.svg\\" class=\\"app-logo\\"/></button></form><div class=\\"jedi-list-item\\">Jar Jar Bings</div><div class=\\"jedi-list-item\\">Anakin Skywalkers</div></div>"`;

exports[`[component] Jedi should render component 1`] = `"<div class=\\"jedi\\"><form id=\\"jedi-form\\" class=\\"jedi-form\\"><div class=\\"jedi-form-field\\"><label class=\\"jedi-form-field-label\\" for=\\"jedi-form-field_name\\">Add a Jedi to the Council</label><input class=\\"jedi-form-field-input\\" id=\\"jedi-form-field_name\\" name=\\"jedi-form-field_name\\" value=\\"\\" placeholder=\\"name\\"/></div><button class=\\"jedi-form-button-submit\\" type=\\"submit\\" title=\\"[disabled] Submit your padawan to the Council\\" disabled=\\"\\"><img alt=\\"app logo\\" src=\\"logo_jedi_order.svg\\" class=\\"app-logo\\"/></button></form><div class=\\"jedi-list-empty\\"><div class=\\"jedi-quote\\"><p>Looks like someone executed Order 66 !</p><blockquote><p>Master sky walker there is too many of them what are we going to do?</p><footer>Padawan</footer></blockquote></div></div></div>"`;

exports[`[component] Jedi should render fetching error 1`] = `"<div class=\\"jedi\\"><form id=\\"jedi-form\\" class=\\"jedi-form\\"><div class=\\"jedi-form-field\\"><label class=\\"jedi-form-field-label\\" for=\\"jedi-form-field_name\\">Add a Jedi to the Council</label><input class=\\"jedi-form-field-input\\" id=\\"jedi-form-field_name\\" name=\\"jedi-form-field_name\\" value=\\"\\" placeholder=\\"name\\"/></div><button class=\\"jedi-form-button-submit\\" type=\\"submit\\" title=\\"[disabled] Submit your padawan to the Council\\" disabled=\\"\\"><img alt=\\"app logo\\" src=\\"logo_jedi_order.svg\\" class=\\"app-logo\\"/></button></form><div class=\\"jedi-list-empty\\"><div class=\\"jedi-quote\\"><p>Looks like someone executed Order 66 !</p><blockquote><p>Master sky walker there is too many of them what are we going to do?</p><footer>Padawan</footer></blockquote></div></div></div>"`;

exports[`[component] Jedi should render fetching error 2`] = `"<div class=\\"jedi error\\"><div class=\\"jedi-error\\"><div class=\\"jedi-quote\\"><p>death star destroyed !</p><blockquote><p>What if I told you that the Republic was now under the control of a Dark Lord of the Sith?</p><footer>Count Dooku</footer></blockquote></div></div><form id=\\"jedi-form\\" class=\\"jedi-form\\"><div class=\\"jedi-form-field\\"><label class=\\"jedi-form-field-label\\" for=\\"jedi-form-field_name\\">Add a Jedi to the Council</label><input class=\\"jedi-form-field-input\\" id=\\"jedi-form-field_name\\" name=\\"jedi-form-field_name\\" value=\\"\\" placeholder=\\"name\\"/></div><button class=\\"jedi-form-button-submit\\" type=\\"submit\\" title=\\"[disabled] Submit your padawan to the Council\\" disabled=\\"\\"><img alt=\\"app logo\\" src=\\"logo_jedi_order.svg\\" class=\\"app-logo\\"/></button></form><div class=\\"jedi-list-empty\\"><div class=\\"jedi-quote\\"><p>Looks like someone executed Order 66 !</p><blockquote><p>Master sky walker there is too many of them what are we going to do?</p><footer>Padawan</footer></blockquote></div></div></div>"`;
